File: samples/Haskell/trigonometric-functions.hs
fromDegrees deg = deg * pi / 180
toDegrees   rad = rad * 180 / pi

example = [
  sin (pi / 6), sin (fromDegrees 30),
  cos (pi / 6), cos (fromDegrees 30),
  tan (pi / 6), tan (fromDegrees 30),
  asin 0.5, toDegrees (asin 0.5),
  acos 0.5, toDegrees (acos 0.5),
  atan 0.5, toDegrees (atan 0.5)]

File: samples/R/trigonometric-functions.r
deg <- function(radians) 180*radians/pi
rad <- function(degrees) degrees*pi/180
sind <- function(ang) sin(rad(ang))
cosd <- function(ang) cos(rad(ang))
tand <- function(ang) tan(rad(ang))
asind <- function(v) deg(asin(v))
acosd <- function(v) deg(acos(v))
atand <- function(v) deg(atan(v))

r <- pi/3
rd <- deg(r)

print( c( sin(r), sind(rd)) )
print( c( cos(r), cosd(rd)) )
print( c( tan(r), tand(rd)) )

S <- sin(pi/4)
C <- cos(pi/3)
T <- tan(pi/4)

print( c( asin(S), asind(S) ) )
print( c( acos(C), acosd(C) ) )
print( c( atan(T), atand(T) ) )
